src/main/java/slimeknights/tconstruct/library/modifiers/modules/technical/MaxArmorLevelModule.java

```diff
@@ -0,0 +1,119 @@
+package slimeknights.tconstruct.library.modifiers.modules.technical;
+
+import net.minecraft.tags.TagKey;
+import net.minecraft.world.entity.EquipmentSlot;
+import net.minecraft.world.entity.EquipmentSlot.Type;
+import net.minecraft.world.entity.LivingEntity;
+import net.minecraft.world.item.Item;
+import net.minecraftforge.common.util.LazyOptional;
+import slimeknights.mantle.data.loadable.Loadables;
+import slimeknights.mantle.data.loadable.primitive.BooleanLoadable;
+import slimeknights.mantle.data.loadable.record.RecordLoadable;
+import slimeknights.tconstruct.library.modifiers.ModifierEntry;
+import slimeknights.tconstruct.library.modifiers.ModifierHooks;
+import slimeknights.tconstruct.library.modifiers.hook.armor.EquipmentChangeModifierHook;
+import slimeknights.tconstruct.library.modifiers.modules.ModifierModule;
+import slimeknights.tconstruct.library.module.HookProvider;
+import slimeknights.tconstruct.library.module.ModuleHook;
+import slimeknights.tconstruct.library.tools.capability.TinkerDataCapability;
+import slimeknights.tconstruct.library.tools.capability.TinkerDataCapability.TinkerDataKey;
+import slimeknights.tconstruct.library.tools.capability.TinkerDataKeys;
+import slimeknights.tconstruct.library.tools.context.EquipmentChangeContext;
+import slimeknights.tconstruct.library.tools.nbt.IToolStackView;
+
+import javax.annotation.Nullable;
+import java.util.List;
+
+/**
+ * Module for keeping track of the total level of a modifier across all pieces of equipment. Does not support incremental, use {@link ArmorStatModule} for that.
+ * @see ArmorStatModule
+ * @see TinkerDataKey
+ * @see slimeknights.tconstruct.library.modifiers.modules.behavior.ShowOffhandModule
+ */
+public record MaxArmorLevelModule(TinkerDataKey<Integer> key, boolean allowBroken, @Nullable TagKey<Item> heldTag) implements HookProvider, EquipmentChangeModifierHook, ModifierModule {
+  private static final List<ModuleHook<?>> DEFAULT_HOOKS = HookProvider.<MaxArmorLevelModule>defaultHooks(ModifierHooks.EQUIPMENT_CHANGE);
+  public static final RecordLoadable<MaxArmorLevelModule> LOADER = RecordLoadable.create(
+    TinkerDataKeys.INTEGER_REGISTRY.requiredField("key", MaxArmorLevelModule::key),
```

Comment:
I am concerned by the usage of `INTEGER_REGISTRY` here, as this module will entirely overwrite any other usage of this key. Our other usages of `INTEGER_REGISTRY` assume only adding is used.

I am inclined to make a new registry for it (same type), though we will have to see how it ends up getting used.